BookendAI Embeddings

stvhu-bookend opened the pull request:
- **Description:** https://bookend.ai/ langchain embeddings integration 
  - **Dependencies:** n/a (just using `requests`)
  - **Tag maintainer:** stvhu-bookend 
  - **Twitter handle:** https://twitter.com/bookendai

stvhu-bookend committed:
BookendAI Embeddings

baskaryan commented on libs/langchain/langchain/embeddings/bookend.py:
```diff
@@ -0,0 +1,84 @@
+    """Bookend AI sentence_transformers embedding models.
+
+    Example:
+        .. code-block:: python
```
need new line between code-block and "from langchain.em..." (otherwise won't render as code block in api docs)

baskaryan commented on libs/langchain/langchain/embeddings/bookend.py:
```diff
@@ -0,0 +1,84 @@
+            bookend.embed_query("She only paints with bold colors; she does not like pastels.")
+    """
+
+    domain: str = Field()
```
nit: don't think you need to add Field here ,can just do `domain: str`

baskaryan commented on libs/langchain/langchain/embeddings/bookend.py:
```diff
@@ -0,0 +1,84 @@
+
+        for text in texts:
+            headers = self.auth_header
+            headers["Content-Type"] = "application/json; charset=utf-8"
```
should this just be done once in init?

stvhu-bookend replied:
thanks, I'll take it out of the loop and just create/initialize it once

baskaryan commented on libs/langchain/langchain/embeddings/bookend.py:
```diff
@@ -0,0 +1,84 @@
+        for text in texts:
+            headers = self.auth_header
+            headers["Content-Type"] = "application/json; charset=utf-8"
+            params = {
```
could be set outside the for loop

baskaryan reviewed (commented):
looks great! you'll want to run `make format` and `make lint` from `libs/langchain` to fix any formatting and linting issues. if you want to give me access to your fork can also fix for you!

stvhu-bookend committed:
Bookend Embeddings PR comments

- addressed `make format`/`make lint`
- fixed code-block api docs formatting
- moved request headers/query params outside of loop in `embed_documents`

stvhu-bookend commented:
> looks great! you'll want to run `make format` and `make lint` from `libs/langchain` to fix any formatting and linting issues. if you want to give me access to your fork can also fix for you!

thanks! I actually didn't know where to run the `make`s before 😅 hopefully the formatting is correct now. addressed your comments in the latest commit!

stvhu-bookend commented:
@baskaryan is there something I need to do to relaunch the Vercel deployment? Unsure why it was cancelled

stvhu-bookend committed:
update to bookend embeddings integ tests

- removed test inputs and stubbed the necessary BookendEmbeddings constructor inputs

stvhu-bookend committed:
Merge branch 'master' into master

stvhu-bookend committed:
Merge branch 'master' into master

stvhu-bookend committed:
Merge branch 'master' into master

stvhu-bookend committed:
Merge branch 'master' into master

stvhu-bookend commented:
@baskaryan friendly bump :)

stvhu-bookend committed:
Merge branch 'master' into master

stvhu-bookend committed:
Merge branch 'master' into master

stvhu-bookend committed:
Merge branch 'master' into master

stvhu-bookend committed:
Merge branch 'master' into master

stvhu-bookend commented:
@baskaryan another friendly bump. hopefully this isn't too annoying 😅

stvhu-bookend committed:
Merge branch 'master' into master

stvhu-bookend commented:
@baskaryan just checking in! wondering should I close and re-open in case this is just lost in the shuffle

stvhu-bookend committed:
Merge branch 'master' into master

stvhu-bookend committed:
Merge branch 'master' into master

stvhu-bookend committed:
Merge branch 'master' into master
